### ISSUE TITLE
[TUI] Initial sync status/progress could be improved

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#3084]
The TUI is not doing a great job of displaying status and progress once we get into "validating chain state" after downloading the txhashset zip file.

It kind of just sits here, slowly creeping up in single digit % for what feels like a _long time_ -  

```
Validating chain state: 8%, step 3/4
```

Then it jumps from around 12% suddenly to step 4/4.

```
Downloading blocks: 99%, step 4/4
```

And here it just stays at 99% the _entire_ time until we are fully synchronized...

I'm' guessing we're downloading less than 1% of the overall blocks, which is great. But not great in terms of sync progress... We should probably use a % of blocks from the sync horizon, not since genesis.

tl;dr

It would be nice to have - 
* Improved % progress during state validation
* Improved % progress during final block download

**Comment by JosephGoulden** [comment]
looking at this.
